pkg/R/DataFrame.R

```diff
@@ -0,0 +1,119 @@
+# DataFrame.R - DataFrame class and methods implemented in S4 OO classes
+
+#' @include jobj.R RDD.R
+NULL
+
+setOldClass("jobj")
+
+#' @title S4 class that represents a DataFrame
+#' @description DataFrames can be created using functions like
+#'              \code{jsonFile}, \code{table} etc.
+#' @rdname DataFrame
+#' @seealso jsonFile, table
+#'
+#' @param env An R environment that stores bookkeeping states of the DataFrame
+#' @param sdf A Java object reference to the backing Scala SchemaRDD
+#' @export
+
+setClass("DataFrame",
+         slots = list(env = "environment",
+                      sdf = "jobj"))
+
+setMethod("initialize", "DataFrame", function(.Object, sdf, isCached, isCheckpointed) {
```

Comment:
`isCached` and `isCheckpointed` doesn't seemed to be used right now. We could remove this and add this back when necessary